src/main/scala/io/altoo/akka/serialization/kryo/KryoSerializer.scala

```diff
@@ -194,56 +183,27 @@ class KryoSerializer(val system: ExtendedActorSystem) extends Serializer {
 
-    // setting default serializer
-    kryo.setDefaultSerializer(defaultSerializerClass)
-
-    // Support serialization of some standard or often used Scala classes
-    kryo.addDefaultSerializer(classOf[scala.Enumeration#Value], classOf[EnumerationSerializer])
-    system.dynamicAccess.getClassFor[AnyRef]("scala.Enumeration$Val") match {
-      case Success(clazz) => kryo.register(clazz)
-      case Failure(e) =>
-        log.error("Class could not be loaded and/or registered: {} ", "scala.Enumeration$Val")
-        throw e
+    serializerType match {
+      case "graph" => kryo.setReferences(true)
+      case "nograph" => kryo.setReferences(false)
+      case o => throw new IllegalStateException("Unknown serializer type: " + o)
     }
-    kryo.register(classOf[scala.Enumeration#Value])
 
-    // identity preserving serializers for Unit and BoxedUnit
-    kryo.addDefaultSerializer(classOf[scala.runtime.BoxedUnit], classOf[ScalaUnitSerializer])
+    val initializer = customSerializerInitClass.getDeclaredConstructor().newInstance()
 
-    // mutable maps
-    kryo.addDefaultSerializer(classOf[scala.collection.mutable.Map[_, _]], classOf[ScalaMutableMapSerializer])
-
-    // immutable maps - specialized by mutable, immutable and sortable
-    kryo.addDefaultSerializer(classOf[scala.collection.immutable.SortedMap[_, _]], classOf[ScalaSortedMapSerializer])
-    kryo.addDefaultSerializer(classOf[scala.collection.immutable.Map[_, _]], classOf[ScalaImmutableMapSerializer])
-
-    // Sets - specialized by mutability and sortability
-    kryo.addDefaultSerializer(classOf[scala.collection.immutable.BitSet], classOf[FieldSerializer[scala.collection.immutable.BitSet]])
-    kryo.addDefaultSerializer(classOf[scala.collection.immutable.SortedSet[_]], classOf[ScalaImmutableSortedSetSerializer])
-    kryo.addDefaultSerializer(classOf[scala.collection.immutable.Set[_]], classOf[ScalaImmutableSetSerializer])
-
-    kryo.addDefaultSerializer(classOf[scala.collection.mutable.BitSet], classOf[FieldSerializer[scala.collection.mutable.BitSet]])
-    kryo.addDefaultSerializer(classOf[scala.collection.mutable.SortedSet[_]], classOf[ScalaMutableSortedSetSerializer])
-    kryo.addDefaultSerializer(classOf[scala.collection.mutable.Set[_]], classOf[ScalaMutableSetSerializer])
-
-    // Map/Set Factories
-    ScalaVersionSerializers.mapAndSet(kryo)
-
-    kryo.addDefaultSerializer(classOf[akka.util.ByteString], classOf[AkkaByteStringSerializer])
-    ScalaVersionSerializers.iterable(kryo)
-    kryo.addDefaultSerializer(classOf[ActorRef], new ActorRefSerializer(system))
-
-    if (settings.kryoTrace)
-      MiniLog.TRACE()
+    // setting default serializer
+    initializer.preInit(kryo)
+    initializer.initScalaSerializer(kryo, system)
+    initializer.initAkkaSerializer(kryo, system)
 
-    kryo.setRegistrationRequired(strategy == "explicit")
+    // if explicit we require all classes to be registered explicitely
+    if (strategy == "explicit")
```

Comment:
for me its hard to read the if blocks: what happens if strategy is `default`?  i would prefer a pattern match here.

Reply:
if anything except `explicit` we don't require registration